whatthepatch/patch.py

```diff
@@ -609,12 +611,11 @@ def parse_unified_diff(text):
                 h = None
                 break
 
-        while len(hunk) > 0:
-            c = unified_change.match(hunk[0])
+        for n in hunk:
+            c = unified_change.match(n)
```

Comment:
Add `hunk.clear()` to preserve original algorithm

Reply:
done

Reply:
Good work